src/main/java/io/jenkins/plugins/monitoring/MonitorPortlet.java

```diff
@@ -3,58 +3,48 @@
 import java.util.Optional;
 
 /**
- * This interface defines the portlet to be shown in the monitoring dashboard.
+ * <p>Implements the {@link IMonitorPortlet} interface and defines a single portlet for the pull request monitoring
+ * dashboard.</p>
+ *
+ * <p>The id must be unique. Please make sure that the id is related to the plugin so that
+ * there are no conflicts with other plugins. It is recommended to use the artifact id
+ * of the plugin or parts of it as prefix. If several portlets of the same class are created in the
+ * {@link MonitorPortletFactory}, it must be ensured that {@link #getId()} is always unique for each portlet!</p>
+ *
+ * @see io.jenkins.plugins.monitoring.IMonitorPortlet
+ * @since 1.6.0
+ * @author Simon Symhoven
  */
-public interface MonitorPortlet {
-
-    /**
-     * Defines the the to be shown.
-     *
-     * @return
-     *          the title.
-     */
-    String getTitle();
-
-    /**
-     * Defines the id for the portlet.
-     *
-     * @return
-     *          the id.
-     */
-    String getId();
-
-    /**
-     * Defines the preferred width of the portlet.
-     *
-     * @return
-     *          the width in pixels.
-     */
-    int getPreferredWidth();
-
-    /**
-     * Defines the preferred height of the portlet.
-     *
-     * @return
-     *          the height in pixels.
-     */
-    int getPreferredHeight();
-
-    /**
-     * Defines the icon to show in the dropdown list of available portlets.
-     *
-     * @return
-     *          the app relative icon url depending on ${resURL} (/static/cache-id),
-     *          or {@code Optional.empty()}, if a default icon should be added.
-     */
-    Optional<String> getIconUrl();
-
-    /**
-     * Defines the relative url to a detail view of the portlet.
-     *
-     * @return
-     *          the relative link to the detail view depending on the current build url
-     *          (e.g. http://localhost:8080/jenkins/job/job-name/view/change-requests/job/PR-1/1/),
-     *          or {@code Optional.empty()}, if no link should be added to portlet.
-     */
-    Optional<String> getDetailViewUrl();
+public abstract class MonitorPortlet implements IMonitorPortlet {
+
+    @Override
+    public String getTitle() {
+        return "Monitor Portlet";
+    }
+
+    @Override
+    public String getId() {
+        return "monitor-portlet";
+    }
+
+    @Override
+    public int getPreferredWidth() {
```

Comment:
I'm not sure if pixels are the right size element. Wouldn't it be easier to use something like the columns concept of Bootstrap?

Reply:
Then, all portlets have to deliver a respsonsive design for the content. In addition, the portlets would then shrink or become larger when the window is reduced or enlarged, which is not necessarily what one wants. Hmm, i am not sure about..

Reply:
Let's keep it this way now and let us discuss that point in our meeting in the afternoon.